### PR TITLE
[NET 6 Attribute Conversion] CoreFoundation

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

**Review by mandel-macaque — APPROVED** [review]

File: src/CoreFoundation/CFBundle.cs [hunk 1]
```diff
@@ -581,21 +581,23 @@ public void UnloadExecutable ()
...
-		[SupportedOSPlatform ("macos11.0"), UnsupportedOSPlatform ("maccatalyst15.0")]
 #endif
 		[DllImport (Constants.CoreFoundationLibrary)]
 		[return: MarshalAs (UnmanagedType.I1)]
 		extern static bool CFBundleIsExecutableLoadable (IntPtr bundle);
 
-#if !NET
+#if NET
+		[SupportedOSPlatform ("macos11.0")]
+		[UnsupportedOSPlatform ("maccatalyst")]
```
Comment (rolfbjarne) [inline src/CoreFoundation/CFBundle.cs:597]: The `[UnsupportedOSPlatform ("ios"), UnsupportedOSPlatform ("tvos")` attributes disasppeared, is that expected?

File: src/CoreFoundation/Dispatch.cs [hunk 2]
```diff
@@ -234,7 +246,9 @@ public DispatchQueue (string label, Attributes attributes, DispatchQueue target
 			}
 		}
 
-#if !NET
+#if NET
+		[SupportedOSPlatform ("ios7.0")]
```
Comment (rolfbjarne) [inline src/CoreFoundation/Dispatch.cs:250]: Would this need a `[SupportedOSPlatform ("tvos")]` to show up as supported on tvOS as well? Or is that coming in a later pass on the attributs?

**Review by rolfbjarne — APPROVED** [review]

File: src/CoreFoundation/CFBundle.cs [hunk 3]
```diff
@@ -581,21 +581,23 @@ public void UnloadExecutable ()
 		}
 
 #if MONOMAC
-#if !NET
+#if NET
+		[SupportedOSPlatform ("macos11.0")]
+		[UnsupportedOSPlatform ("maccatalyst")]
```
Comment (tj-devel709) [inline src/CoreFoundation/CFBundle.cs:586]: Looks like the maccatalyst version numbers fell off here as well. Was that intended?

File: src/CoreFoundation/CFBundle.cs (hunk 1 above)
Comment (chamons) [reply src/CoreFoundation/CFBundle.cs:597]: I believe so, CFBundle doesn't have attributes anywhere else FWICT:

```
donblas@Masterwork src % git grep CFBundle | grep interface    
donblas@Masterwork src % git grep CFBundle | grep class        
...
	public partial class CFBundle {
...
donblas@Masterwork src % head -n 25 CoreFoundation/CFBundle.cs 
...
	public partial class CFBundle : NativeObject {
```

and the only legacy attributes are:

```
		[Introduced (PlatformName.MacOSX, 11, 0)]
		[NoMacCatalyst]
```

File: src/CoreFoundation/CFBundle.cs (hunk 3 above)
Comment (chamons) [reply src/CoreFoundation/CFBundle.cs:586]: I believe so, it is mirroring ` 		[NoMacCatalyst]`.

**Review by rolfbjarne — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by chamons** [comment]
Test failure https://github.com/xamarin/maccore/issues/2443

**Pull request merged**